### PR TITLE
Small fixes for windows genlink and -mcpu flag

### PR DESCRIPTION
**Pull request by LarsGardien**
- -mcpu flag was not getting set correctly for some cortex-m* MCU's

- Spaces in path to genlink.py could break the call to genlink.py
- Calling $(shell genlink.py) resolved to the script's shebang value /usr/bin/env python. The /usr/bin/env command is not available for some windows users

### CODE REVIEW

File: ld/devices.data [hunk 1]
```diff
@@ -451,9 +451,9 @@ lpc43xx_m4 lpc43xx CPU=cortex-m4 FPU=hard-fp4-sp-d16
 # the STM32 families
 
 stm32f0 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m0 FPU=soft
-stm32f1 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m3 FPU=soft
-stm32f2 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m3 FPU=soft
-stm32f3 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m4 FPU=hard-fpv4-sp-d16
+stm32f1 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m1 FPU=soft
+stm32f2 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m2 FPU=soft
+stm32f3 END ROM_OFF=0x08000000 RAM_OFF=0x20000000 CPU=cortex-m3 FPU=hard-fpv4-sp-d16
```
Comment (karlp) [inline ld/devices.data]: no.  please elaborate what this is meant to be fixing.  There is no such thing as a cortex-m2, and cortex-m1 is the fpga implementation.

File: mk/genlink-config.mk [hunk 2]
```diff
@@ -24,11 +24,11 @@ endif
...
-genlink_family		:=$(shell $(OPENCM3_DIR)/scripts/genlink.py $(DEVICES_DATA) $(DEVICE) FAMILY)
-genlink_subfamily	:=$(shell $(OPENCM3_DIR)/scripts/genlink.py $(DEVICES_DATA) $(DEVICE) SUBFAMILY)
-genlink_cpu		:=$(shell $(OPENCM3_DIR)/scripts/genlink.py $(DEVICES_DATA) $(DEVICE) CPU)
-genlink_fpu		:=$(shell $(OPENCM3_DIR)/scripts/genlink.py $(DEVICES_DATA) $(DEVICE) FPU)
-genlink_cppflags	:=$(shell $(OPENCM3_DIR)/scripts/genlink.py $(DEVICES_DATA) $(DEVICE) CPPFLAGS)
+genlink_family		:=$(shell python "$(OPENCM3_DIR)/scripts/genlink.py" $(DEVICES_DATA) $(DEVICE) FAMILY)
+genlink_subfamily	:=$(shell python "$(OPENCM3_DIR)/scripts/genlink.py" $(DEVICES_DATA) $(DEVICE) SUBFAMILY)
+genlink_cpu		:=$(shell python "$(OPENCM3_DIR)/scripts/genlink.py" $(DEVICES_DATA) $(DEVICE) CPU)
+genlink_fpu		:=$(shell python "$(OPENCM3_DIR)/scripts/genlink.py" $(DEVICES_DATA) $(DEVICE) FPU)
+genlink_cppflags	:=$(shell python "$(OPENCM3_DIR)/scripts/genlink.py" $(DEVICES_DATA) $(DEVICE) CPPFLAGS)
```
Comment (karlp) [inline mk/genlink-config.mk]: hrm, this is .... hairy.  I'll need more windows testing on this, and unfortunately it also has a lot of potential to break other systems, where "python" is no longer clear what it is, python2 or python3.    Are you building with msys-git? or inside cmd.exe itself?

Comment (karlp) [reply mk/genlink-config.mk]: (the _quoting_ is unnecessary without the python call)

File: ld/devices.data (hunk 1 above)
Comment (LarsGardien) [reply ld/devices.data]: You're right. I thought it fixed some issues for me but it didn't. Correlation not causation...

File: mk/genlink-config.mk (hunk 2 above)
Comment (LarsGardien) [reply mk/genlink-config.mk]: I'm running it using atollic truestudio, which uses a built-in make.exe using cmd if i'm correct. This just fixed it for me. Thought I'd share my two cents. 

"it also has a lot of potential to break other systems". Definitely true.

Comment (karlp) [reply mk/genlink-config.mk]: Did you try with _only_ the quoting?

**Comment by karlp** [comment]
Setting OPENCM3_DIR to a full path with spaces in it causes problems on linux too, and needs more fixed than just the quoting you've got.  Try using a relative path that hopefully doesn't include spaces, or just avoiding spaces altogether for starters.   Do you then _still _ have problems with the script execution?

**Comment by karlp** [comment]
Any updates? I'd _like_ windows support to work, but I'd like to resolve which bits are _necessary_ rather than just a "works for me" approach.

**Comment by karlp** [comment]
Please re-open with more information if you are still working on this.

**Pull request closed without merging**